TACC/Core-CMS-Custom
# Task/WP 152 extension admin table uth requests

## Overview
Updated Extension Admin Table per UT Health's requests 

## Related

- [WP-152](https://jira.tacc.utexas.edu/browse/WP-152)
- duplicated by [WP-108](https://jira.tacc.utexas.edu/browse/WP-108)

## Changes

Added the following: 
- status filter
- organization filter
- styles for filters to make them mobile friendly
- added styles for extension modal and table
- Created success and error templates for edit extension modal
- Add action drop down
a. View Extension 
> 1. removed duplicate entries
> 2. Reordered entries similar to the table

b. Edit Extension modal allows edit of:
> 1. Change applicable data period
> 2. Outcome (pulling entries directly from DB no hardcoded values)
> 3. Status (pulling entries directly from DB no hardcoded values)
> 4. Notes (pulls notes from DB into the field to add or edit)
> 5. All entries are optional. If left blank, does not write over what is in DB.

## Testing

1. Reconfigure your settings in taccsite_cms/secrets.py to use the Stache test DB credentials for APCD_DATABASE = {}. This way you will not have to use hardcoded test info anymore.
2. Go to [http://localhost:8000/administration/list-extensions/](http://localhost:8000/administration/list-extensions/)
3. Check that the filters work
4. Shrink the page / check mobile view and make sure the filters don't overlap. Make sure the column headers match to the right values. When page shrinks, columns switch to rows. Make sure they align.
5. Check out View Extension modal to make sure email address is no longer listed twice and mimics the order of the table
6. Check the edit modal. All fields are optional. Make sure you can submit and that only the fields that aren't empty are updated
7. Check that all fields are able to be updated and set correctly from the modal


## UI
[Main page](http://localhost:8000/administration/list-extensions/)

![Screenshot 2023-06-26 at 5 08 46 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/7fc82e94-8131-4983-92cd-f9010c000587)


[Status and Org filtered ](http://localhost:8000/administration/list-extensions/?status=Pending&org=Tacc%20Test)
![Screenshot 2023-06-26 at 5 09 25 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/7e3d7219-613e-4503-b6df-037acfcea4c1)


View Extension modal
![Screenshot 2023-06-26 at 5 09 56 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/098a16af-b6f0-430c-8f01-d64b2b43908a)



Edit Extension modal
![Screenshot 2023-06-26 at 5 10 09 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/950714bd-9b5b-4e55-86fe-ed737a22aee4)

Before Edit
![Screenshot 2023-06-26 at 5 10 34 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/db627345-538d-422d-b041-4d89b000c510)

After successful edit
![Screenshot 2023-06-26 at 5 11 43 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/6c095895-98f2-4df4-94c8-1b7a9353d4c3)

Edit Success page
![Screenshot 2023-06-26 at 5 12 14 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/4a49225c-8776-49df-ab47-f1498cf2ba6e)
